src/Streamly/Coreutils/Uniq.hs

```diff
@@ -0,0 +1,215 @@
+module Streamly.Coreutils.Uniq (
+        Output (..)
+      , UniqResult (..)
+      , UniqOptions (..)
+      , defaultUniqOptions
+      , getRepetition
+      , compareUsingOptions
+      , uniq
+   )
+where
+import qualified Streamly.Prelude as S
+import qualified Streamly.Internal.Data.Fold as FL
+
+import Data.Char (toLower, isSpace)
+import Streamly.Internal.Data.Stream.StreamK.Type (IsStream)
+
+
+-- |
+-- Data type to capture the output of the stream -
+-- the stream should either be composed of
+-- unique, repeated or duplicate strings
+--
+-- @since 0.1.0.0
+data Output = Unique
```

Comment:
You can maybe add `All` to this that would just be `getRepetition`

Reply:
okay. :+1: good idea